src/pages/Compound/useFarmBotRewards.ts

```diff
@@ -0,0 +1,48 @@
+import { useContractKit } from '@celo-tools/use-contractkit'
+import { useCallback, useEffect, useState } from 'react'
+import { AbiItem } from 'web3-utils'
+
+import farmBotAbi from '../../constants/abis/FarmBot.json'
+import { useFarmRegistry } from '../Earn/useFarmRegistry'
+
+export interface FarmBotRewards {
+  address: string
+  stakingRewardsAddress: string
+  rewardsUSDPerYear?: string
+  tvlUSD?: string
+}
+
+export const useFarmBotRewards = (farmBotAddresses: string[]) => {
```

Comment:
Why the decision to split farmbot rewards info into a new method? All farmbots have reward information associated with them. It looks like we're not getting reward info for metafarms. APY calculations for metafarms should be derived from the `FarmBotRewards` information of both the meta-RFP and the underlying RFP.

Reply:
i decided to split the farmbot info and rewards because the meta farms need a different way to calculate the rewards to include the rewards from the underlying RFPs. i was thinking something like the meta farms rewards hook could use this hook + add some logic.

i could have gone with the approach of keeping all the logic in the hook ( e.g. `useFarmBotRegistry(farmbotType)` where `farmbotType` is an enum or something) but then the hook kind of diverges on one part of the logic and has to manage 2 sets of addresses. i felt the hook was getting a bit big and scary, so thought to keep the farmbot/meta farmbot addresses with the zap/provide functionalities but i guess this is pretty subjective. do you have a strong preference?

Reply:
Ahh, gotcha -- that makes sense to me! I'm fine with this approach